src/osd/OSD.h

```diff
@@ -670,26 +675,30 @@ class OSDService : public Scrub::ScrubSchedListener {
     std::lock_guard l(recovery_lock);
     awaiting_throttle.remove_if(
       [pg](decltype(awaiting_throttle)::const_reference awaiting ) {
-	return awaiting.second.get() == pg;
+        return awaiting.pg.get() == pg;
       });
   }
 
   unsigned get_target_pg_log_entries() const;
 
   // delayed pg activation
-  void queue_for_recovery(PG *pg) {
+  void queue_for_recovery(PG *pg, uint64_t cost_per_object) {
     std::lock_guard l(recovery_lock);
 
     if (pg->is_forced_recovery_or_backfill()) {
-      awaiting_throttle.push_front(std::make_pair(pg->get_osdmap()->get_epoch(), pg));
+      awaiting_throttle.emplace_front(
+        pg->get_osdmap()->get_epoch(), pg, cost_per_object);
     } else {
-      awaiting_throttle.push_back(std::make_pair(pg->get_osdmap()->get_epoch(), pg));
+      awaiting_throttle.emplace_back(
+        pg->get_osdmap()->get_epoch(), pg, cost_per_object);
```

Comment:
@athanatos The initialization of pg_awaiting_throttle_t is missing. Will add it in the next push.

Reply:
Fixed in the latest push.